src/equisolve/numpy/preprocessing/_base.py

```diff
@@ -205,7 +188,7 @@ def fit(
                     [Labels.single()],
                 )
 
-                if self.with_std:
+                if self.with_std and (parameter in self.parameter_keys):
                     X_mean = np.average(X_mat, weights=sample_weights, axis=0)
                     var = np.average((X_mat - X_mean) ** 2, axis=0)
```

Comment:
Maybe it is actually useful to use the `mean_over_samples` and `variance_over_samples` function here.

Reply:
adding to issue https://github.com/lab-cosmo/equisolve/issues/18